devdoc/interlocked_linux_requirements.md

```diff
@@ -0,0 +1,345 @@
+interlocked linux
+================
+
+## Overview
+
+`interlocked linux` is the Linux-specific implementation of the `interlocked` module. Each function calls the corresponding `atomic` function from `stdatomic.h`.

```

Comment:
>`atomic` function from `stdatomic.h`. [](start = 124, length = 37)

Should add a note that this is always using the memory order `memory_order_seq_cst` (I believe that's the default for all of the calls when you don't call the _explicit version and specify).

This should be done to match the Interlocked win32 api which takes a full memory barrier for each operation.